sql/src/main/java/org/apache/druid/sql/calcite/planner/PlannerContext.java

```diff
@@ -559,6 +559,14 @@ public void addAllToQueryContext(Map<String, Object> toAdd)
     initializeContextFields();
   }
 
+  /**
+   * Add additional query context parameters to planner config, overriding any existing values.
+   */
+  public void addAllToPlannerConfig(Map<String, Object> toAdd)
+  {
+    this.plannerConfig = this.plannerConfig.withOverrides(toAdd);
+  }
+
```

Comment:
How about just adding this logic to `initializeContextFields` or `addAllToQueryContext` instead of a new method? It seems like callers would always want to call both methods, so I don't see a good reason to keep them separate since it would make it easier to make mistakes.

To go with that, I think we could also adjust the `PlannerContext` constructor to no longer take the `PlannerConfig` argument since we can get it from the `PlannerToolbox`, and we don't need the `Preconditions.checkNotNull` that is currently in the constructor because `PlannerToolbox` also has that check, making the one in this constructor redundant. This would mean that `PlannerContext.create` would no longer need to also be calling `withOverrides` on the context before passing to the constructor

Reply:
Yeah, makes sense to add it to `initializeContextFields`. I've also renamed this private method to `initializeContextFieldsAndPlannerConfig`